source/extensions/filters/common/ext_authz/ext_authz_impl.cc

```diff
@@ -47,23 +50,50 @@ void GrpcClientImpl::check(RequestCallbacks& callbacks,
 
 void GrpcClientImpl::onSuccess(
     std::unique_ptr<envoy::service::auth::v2alpha::CheckResponse>&& response, Tracing::Span& span) {
-  CheckStatus status = CheckStatus::OK;
   ASSERT(response->status().code() != Grpc::Status::GrpcStatus::Unknown);
+  ResponsePtr authz_response = std::make_unique<Response>(Response{});
+
   if (response->status().code() != Grpc::Status::GrpcStatus::Ok) {
-    status = CheckStatus::Denied;
     span.setTag(Constants::get().TraceStatus, Constants::get().TraceUnauthz);
+    authz_response->status = CheckStatus::Denied;
+    authz_response->status_code = enumToInt(Http::Code::Forbidden);
+
+    if (response->has_denied_response()) {
+      for (const auto& header : response->denied_response().headers()) {
```

Comment:
There's some missing coverage, check out https://49526-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.source_extensions_filters_common_ext_authz_ext_authz_impl.cc.html.

Reply:
Haven't addressed all the tests yet. Will do.